src/main.py

```diff
@@ -2035,13 +2043,34 @@ def _do_primary_switchover(self, switchover_candidate, zk_state):
 
         return True
 
-    def _candidate_is_sync_with_primary_with_get_state(self, switchover_candidate):
-        db_state = self.db.get_state()
-        return self._candidate_is_sync_with_primary(db_state, switchover_candidate)
+    def _wait_candidate_is_sync_with_primary(self, switchover_candidate, timeout=60, max_attempts=5):
+        """
+        We waiting for a short period of time for candidate to catchup (replay).
+        We use linear wait here to minimize delay in positive scenario.
+        Old primary may experience short load spikes after closing bouncer
+        (for unknown yet reason), so we are trying to reconnect few times.
+        But if we failed to reconnect we treat local postgres as dead and
+        continue switchover.
+        """
+        deadline = time.time() + timeout
+        attempt = 0
+        while time.time() < deadline:
+            replics_info = self.db.get_replics_info('primary')
+            if replics_info is None:
```

Comment:
It can't be None.
`pg.get_replics_info` either returns list (maybe empty): https://github.com/yandex/pgconsul/blob/main/src/pg.py#L103
or raises exception: https://github.com/yandex/pgconsul/blob/main/src/pg.py#L97